.github/workflows/cron.yml

```diff
@@ -0,0 +1,338 @@
+      # ${lib_name}.diff file for every outdated library - for output in the issue
+      - name: Diff public API changes between outdated libs
+        if: env.ISSUE_ALREADY_EXISTS == 'false' && env.LIBS_UP_TO_DATE == 'false' && env.BUILD_FAILING == 'true'
+        env:
+          CARGO_TERM_COLOR: always
+        run: |
+          IFS=';' read -ra arr <<< "$OUTDATED_LIBS"
+
+          # we use this program for grepping code contextually, from within the rust tool
+          cargo install ast-grep
+
+          for lib_name in "${arr[@]}"; do
+            if [[ -z "$lib_name" ]]; then
+              continue
+            fi
+
+            # this is faster than "cargo outdated", especially in a loop
+            LIB_LATEST_VERSION=$(curl --silent "https://crates.io/api/v1/crates/$lib_name" | jq -r '.crate.max_stable_version')
+            LIB_CURRENT_VERSION=$(cargo metadata --format-version=1 -q --manifest-path=./uniffi-zcash-lib/lib/Cargo.toml | jq -r --arg lib_name "$lib_name" '.packages[] | select(.name == $lib_name) | .version')
+
+            # write the diffs to files, which we show in a separate step for better readability
+            # for colored output ANSI color codes are written in the file and can't be rendered in markdown
+            #
+            # colored output
+            cargo run \
+            --manifest-path=./uniffi-zcash-lib/lib/Cargo.toml \
+            -p uniffi-zcash-cli diff \
+            --grep-dir `realpath ./uniffi-zcash-lib/lib/uniffi-zcash` \
+            --lib-name "$lib_name" \
+            --lib-old-version "$LIB_CURRENT_VERSION" \
+            --lib-new-version "$LIB_LATEST_VERSION" \
+            --color always > "${lib_name}_colored.diff"
+
+            # non-colored output
+            cargo run \
+            --manifest-path=./uniffi-zcash-lib/lib/Cargo.toml \
+            -p uniffi-zcash-cli diff \
+            --grep-dir `realpath ./uniffi-zcash-lib/lib/uniffi-zcash` \
+            --lib-name "$lib_name" \
+            --lib-old-version "$LIB_CURRENT_VERSION" \
+            --lib-new-version "$LIB_LATEST_VERSION" \
+            --color never > "${lib_name}.diff"
+          done
+
+      # Step only for better diff visibility and linkability
+      # If you change the name of this step, make sure it's changed in the "Create workflow summary" gh command too
+      - name: Show public API diffs
+        if: env.ISSUE_ALREADY_EXISTS == 'false' && env.LIBS_UP_TO_DATE == 'false' && env.BUILD_FAILING == 'true'
+        run: |
+          IFS=';' read -ra arr <<< "$OUTDATED_LIBS"
+
+          for lib_name in "${arr[@]}"; do
+            if [[ -z "$lib_name" ]]; then
+              continue
+            fi
+            echo "---"
+            echo "${lib_name}"
+            echo "---"
+            cat "${lib_name}_colored.diff"
+            echo ""
+          done
+
+      - name: Create workflow summary
```

Comment:
I use the summary to show the full output, because it seems that it has no body size limit (or at least I haven't reached it)